packages/docusaurus-theme-common/src/utils/useLocationChange.ts

```diff
@@ -21,10 +22,12 @@ export function useLocationChange(onLocationChange: OnLocationChange): void {
   const location = useLocation();
   const previousLocation = usePrevious(location);
 
+  const onLocationChangeDynamic = useDynamicCallback(onLocationChange);
+
   useEffect(() => {
-    onLocationChange({
+    onLocationChangeDynamic({
```

Comment:
@slorber I just noticed that this change negatively affects on the hideable sidebar feature. After location change (I suppose), while closing a sidebar, the current scroll position is reset to zero, i.e. a page scrolls to the top.

Actual behavior:

https://user-images.githubusercontent.com/4408379/144337991-1c64d461-fe68-4d1a-aba4-d975ced046b7.mp4

Expected behavior

https://user-images.githubusercontent.com/4408379/144338188-8f97746c-e145-4831-8c44-7291aec7a2ad.mp4

Could you please see how this can be fixed?